python/cudf/cudf/tests/text/test_text_methods.py

```diff
@@ -855,3 +855,47 @@ def test_minhash():
     with pytest.raises(ValueError):
         seeds = cudf.Series([0, 1, 2], dtype=np.int32)
         strings.str.minhash(seeds=seeds)
+
+
+def test_jaccard_index():
+    str1 = cudf.Series(["the brown dog", "jumped about"])
+    str2 = cudf.Series(["the black cat", "jumped around"])
+
+    expected = cudf.Series([0.058824, 0.307692], dtype=np.float32)
+    actual = str1.str.jaccard_index(str2, 5)
+    assert_eq(expected, actual)
+
+    actual = str2.str.jaccard_index(str1, 5)
+    assert_eq(expected, actual)
+
+    # adopted from https://github.com/rapidsai/rapids-deduplication/issues/36
+    da = str1.str.character_ngrams(5, True)
+    db = str2.str.character_ngrams(5, True)
+    da = da.list.unique()
+    db = db.list.unique()
+    da = da.explode()
+    db = db.explode()
+    da = da.to_frame()
+    db = db.to_frame()
+    da = da.reset_index()
+    db = db.reset_index()
+    da = da.rename(columns={0: "token"})
+    db = db.rename(columns={0: "token"})
+    db["match"] = 1
+    inter = da.merge(db, on=["index", "token"], how="left")
+    inter = inter.groupby("index")["match"].sum()
+    union = da.merge(db, on=["index", "token"], how="outer")
+    union = union.groupby("index").size()
+    res = inter / union
+    res = res.sort_index()
+    res = res.values.astype("float32")
+    expected = cudf.Series(res)
```

Comment:
Thanks! This is helpful. However, my suggestion would be to run this in a separate pytest on some collections of random and longer strings rather than the same strings that you've already validated explicitly in the same test.

Reply:
Quick suggestion for how this could be generated:
```
import random, string, cudf

word_length = 5
num_words = 10
s = cudf.Series([''.join(random.choice(string.ascii_lowercase) for _ in range(word_length)) for _ in range(num_words)])
```

Reply:
Sorry, I've run out of time now. I'm not going to be able to do this before the end of burndown.
If you want, you can move this PR to 23.10 when that gets created later this week.
Thanks for reviewing!

Reply:
I added this test myself and verified that it works locally.